src/plugins/plot/chart/MctChart.vue

```diff
@@ -279,10 +279,11 @@ export default {
       this.makeChartElement(series);
       this.makeLimitLines(series);
     },
-    onSeriesRemove(series) {
+    onSeriesRemove(series, index) {
       this.stopListening(series);
       this.removeChartElement(series);
       this.scheduleDraw();
+      this.seriesModels.splice(index, 1);
```

Comment:
Hmm. After series reordering occurs the incorrect series could get removed here if we use the `index`.
Consider using the following approach instead of using the `index`
https://github.com/nasa/openmct/pull/6961/files#diff-ff5740a785ad44ad9fcb7df262125bf8857a763020881fe65802a8aae49da8aaR205